google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1beta2/CivilTimeEncoderTest.java

```diff
@@ -0,0 +1,425 @@
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.threeten.bp.LocalDateTime;
+import org.threeten.bp.LocalTime;
+
+@RunWith(JUnit4.class)
+public class CivilTimeEncoderTest {
+  private static final Logger LOG = Logger.getLogger(CivilTimeEncoderTest.class.getName());
+
+  @Test
+  public void Packed64TimeMicrosTest() {
+    // 00:00:00.000000
+    // 0b000000000000000000000000000|00000|000000|000000|00000000000000000000
+    // 0x0
+    assertEquals(0x0L, CivilTimeEncoder.encodePacked64TimeMicros(LocalTime.of(0, 0, 0, 0)));
+    // 00:01:02.003000
+    // 0b000000000000000000000000000|00000|000001|000010|00000000101110111000
+    // 0x4200BB8
+    assertEquals(
+        0x4200BB8L, CivilTimeEncoder.encodePacked64TimeMicros(LocalTime.of(0, 1, 2, 3_000_000)));
+    // 12:00:00.000000
+    // 0b000000000000000000000000000|01100|000000|000000|00000000000000000000
+    // 0xC00000000
+    assertEquals(
+        0xC00000000L, CivilTimeEncoder.encodePacked64TimeMicros(LocalTime.of(12, 0, 0, 0)));
+    // 13:14:15.016000
+    // 0b000000000000000000000000000|01101|001110|001111|00000011111010000000
+    // 0xD38F03E80
+    assertEquals(
+        0xD38F03E80L,
+        CivilTimeEncoder.encodePacked64TimeMicros(LocalTime.of(13, 14, 15, 16_000_000)));
+    // 23:59:59.999000
+    // 0b000000000000000000000000000|10111|111011|111011|11110011111001011000
+    // 0x17EFBF3E58
+    assertEquals(
+        0x17EFBF3E58L,
+        CivilTimeEncoder.encodePacked64TimeMicros(LocalTime.of(23, 59, 59, 999_000_000)));
+
+    // 00:00:00.000000
+    // 0b000000000000000000000000000|00000|000000|000000|00000000000000000000
+    // 0x0
+    assertEquals(LocalTime.of(0, 0, 0, 0), CivilTimeEncoder.decodePacked64TimeMicros(0x0L));
+    // 00:01:02.000003
+    // 0b000000000000000000000000000|00000|000001|000010|00000000000000000011
+    // 0x4200003
+    assertEquals(
+        LocalTime.of(0, 1, 2, 3_000), CivilTimeEncoder.decodePacked64TimeMicros(0x4200003L));
+    // 12:00:00.000000
+    // 0b000000000000000000000000000|01100|000000|000000|00000000000000000000
+    // 0xC00000000
+    assertEquals(
+        LocalTime.of(12, 0, 0, 0), CivilTimeEncoder.decodePacked64TimeMicros(0xC00000000L));
+    // 13:14:15.000016
+    // 0b000000000000000000000000000|01101|001110|001111|00000000000000010000
+    // 0xD38F00010
+    assertEquals(
+        LocalTime.of(13, 14, 15, 16_000), CivilTimeEncoder.decodePacked64TimeMicros(0xD38F00010L));
+    // 23:59:59.999999
+    // 0b000000000000000000000000000|10111|111011|111011|11110100001000111111
+    // 0x17EFBF423F
+    assertEquals(
+        LocalTime.of(23, 59, 59, 999_999_000),
+        CivilTimeEncoder.decodePacked64TimeMicros(0x17EFBF423FL));
+
+    try {
+      // 00:00:00.000000
+      // 0b000000000000000000000000001|00000|000000|000000|00000000000000000000
+      // 0x2000000000
+      CivilTimeEncoder.decodePacked64TimeMicros(0x2000000000L);
+      Assert.fail();
```

Comment:
Is this case suppose to pass or fail?

Reply:
Yes all of the ones with assert.fail() are pulled from the zetasql tests and are passing in invalid bits/too large numbers/etc and are meant to fail so that a user can't decode a time that doesn't exist.

Reply:
I still don't quite understand, so in this case, it is suppose to fail and the error doesn't have a message? Why this is a failing case? Same question for the case below.

Reply:
I've added assert.fail() to all other tests that are intended to fail for more clarity